### PR TITLE
Add API to create view controllers without registering features

### PR DESCRIPTION
**Pull request by JohnSundell**
This change adds a new way to create view controllers, without having to pre-register a feature with the framework. The use case for this is to more easily be able to create view controller, which decreases the learning curve to get into using the framework.

It also will make it easier to use the Hub Framework for playgrounds.

To make this happen, another convenience API is introduced to be able to create content operation factories using a concrete class that takes a block. Again, this reduces the barrier of entry for creating content operations, and the need for boilerplate.

### CODE REVIEW

**Comment by JohnSundell** [comment]
@spotify/objc-dev

**Comment by JohnSundell** [comment]
Also planning to update the "Getting Started Guide" to use this API instead, since it's so much easier to get into and doesn't require modifying the navigation stack. Will do in a subsequent PR thou :)

File: include/HubFramework/HUBViewControllerFactory.h [hunk 1]
```diff
@@ -46,17 +47,37 @@ NS_ASSUME_NONNULL_BEGIN
...
+ *         Hub Framework's feature registry, it will simply be assigned to the view controller.
+ *  @param contentOperations The content operations to use to load the content for the view controller.
+ *  @param featureIdentifier The identifier of the feature that the view controller will belong to. Will
+ *         be made available to content operations as part of `HUBFeatureInfo`.
+ *  @param featureTitle The title of the feature that the view controller will belong to. Used for its
+ *         default title, and also made available to contnet operations as part of `HUBFeatureInfo`.
+ */
+- (UIViewController<HUBViewController> *)createViewControllerForViewURI:(NSURL *)viewURI
+                                                      contentOperations:(NSArray<id<HUBContentOperation>> *)contentOperations
+                                                      featureIdentifier:(NSString *)featureIdentifier
```
Comment (Goos) [inline include/HubFramework/HUBViewControllerFactory.h]: If the point of this method is to enable creating view controllers without registering a feature, does it still make sense to require a feature identifier?

File: include/HubFramework/HUBBlockContentOperationFactory.h [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+@interface HUBBlockContentOperationFactory : NSObject  <HUBContentOperationFactory>
+
+/**
+ *  Initialize an instance of this class with a block that creates content operations
+ *
+ *  @param block The block used to create content operations. The input parameter of the block will
+ *         be the view URI that content operations should be created for. This block will be copied
+ *         and called every time this factory is asked to create content operations.
+ */
+- (instancetype)initWithBlock:(NSArray<id<HUBContentOperation>> *(^)(NSURL *))block HUB_DESIGNATED_INITIALIZER;
```
Comment (Goos) [inline include/HubFramework/HUBBlockContentOperationFactory.h]: Perhaps we could typedef this block to make it more readable?

File: include/HubFramework/HUBViewControllerFactory.h (hunk 1 above)
Comment (JohnSundell) [inline include/HubFramework/HUBViewControllerFactory.h]: Yeah, it's needed for other APIs, such as passing `HUBFeatureInfo` to content operations. Better to require it to enable content operations to be reusable between view controllers created this way and those created from feature registrations.

**Comment by Goos** [comment]
Awesome stuff! 🎉

File: include/HubFramework/HUBBlockContentOperationFactory.h (hunk 2 above)
Comment (JohnSundell) [inline include/HubFramework/HUBBlockContentOperationFactory.h]: Good suggestion, I'll do that! 👍

File: tests/HUBViewControllerFactoryTests.m [hunk 3]
```diff
@@ -103,6 +103,30 @@ - (void)testCreatingViewControllerForInvalidViewURIReturnsNil
...
+    UIViewController<HUBViewController> * const viewController = [self.manager.viewControllerFactory createViewControllerForViewURI:viewURI
+                                                                                                                  contentOperations:@[contentOperation]
+                                                                                                                  featureIdentifier:@"identifier"
+                                                                                                                       featureTitle:@"Title"];
+    
+    [viewController viewWillAppear:NO];
+    
+    XCTAssertTrue(contentOperationCalled);
+    XCTAssertEqualObjects(viewController.featureIdentifier, @"identifier");
+    XCTAssertEqualObjects(viewController.navigationItem.title, @"Title");
```
Comment (nataliq) [inline tests/HUBViewControllerFactoryTests.m]: What about checking the feature registry for a registration for the provided URI?

Comment (nataliq) [inline tests/HUBViewControllerFactoryTests.m]: Never mind, I was under the impression that there is a feature registration behind the scenes and my suggestion was checking that it is not nil. My bad :)

**Pull request merged**